src/renderer/widgets/CreateWallet/model/signatory-model.ts

```diff
@@ -31,6 +32,16 @@ sample({
   target: $signatories,
 });
 
+sample({
```

Comment:
Can you use simple map here?
```ts
const $hasDuplicateSignatories = $signatories.map((signatories) => {
    const signatoriesArray = Array.from(signatories.values()).map(({ address }) => toAccountId(address));

    return new Set(signatoriesArray).size !== signatoriesArray.length;
  })
```

Reply:
I can't because `$signatories` is a `Map`. so that we keep the order (there's no guarantee in arrays).

Reply:
or do you mean, I shouldn't bother with sample, and directly derive the store. This make sense. Let me change

Reply:
I used a combine now in [34765f6](https://github.com/novasamatech/nova-spektr/pull/2518/commits/34765f6bbc0a66dd7f9c2f79262b8167009698fe)

```ts
const $hasDuplicateSignatories = combine($signatories, (signatories) => {
  const signatoriesArray = Array.from(signatories.values()).map(({ address }) => toAccountId(address));

  return new Set(signatoriesArray).size !== signatoriesArray.length;
});
```